### DISCUSSION TITLE
How to change the log_name 'default'?

### DISCUSSION THREAD

**Discussion by kusman28** [spatie/laravel-activitylog#724] [Q&A] [answered]
I want to change the value of log_name 'default' to the user that has made the changes. 
something like
`auth()->user()->name`
![image](https://user-images.githubusercontent.com/40085614/80920839-fbfc7280-8da4-11ea-81f5-88614f2eb3f6.png)

**Comment by Gummibeer** [comment, marked as answer]
Hey,

if you have an instance of the `ActivityLogger` you can use one of these two methods to define it on runtime.
https://github.com/spatie/laravel-activitylog/blob/469e78d6dd38baa7ef3a602f238f979afe2a4000/src/ActivityLogger.php#L115-L125
If you want to set the log name on model event logs you can use the `tapActivity()` method on each model to set the `log_name` attribute of the `Activity` model instance passed in.
The third option would be to use the `Macroable` trait of `ActivityLogger` class and add a method that can override the `defaultLogName` property on runtime. This will change the log name for every activity without a defined one after the method is called.

At all I don't recommend to do this. Because the causing user is already saved as a relationship on the model and the log name/channel is more for different areas of your app for example. So like admin and user logs.
Or backoffice and API or whatever you have.

**Reply by kuvar0158** [reply]
ok good

**Reply by kuvar0158** [reply]
protected static $logName = 'user';  
you can also add this line on users model
